examples/mongocxx/get_values_from_documents.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/client.hpp>
+
+using namespace bsoncxx;
+
+// Document model, showing array with nested documents:
+//
+// {
+//     "_id" : ObjectId(...),
+//     "messagelist" : [
+//       {
+//         "uid" : 413098706,
+//         "status" : 2,
+//         "msg": "..."
+//       },
+//       ...
+//     ]
+// }
+
+// Construct a document in the format of 'messagelist'
+document::value new_message(int64_t uid, int32_t status, std::string msg) {
+    builder::stream::document doc;
+    return doc << "uid" << uid << "status" << status << "msg" << msg << builder::stream::finalize;
+}
+
+// Insert a document into the database
+void insert_test_data(mongocxx::collection& coll) {
```

Comment:
Suggest changing `coll` to be a pointer type, instead of a reference type.  This is just straight from the Google C++ style guide; see <https://google.github.io/styleguide/cppguide.html#Reference_Arguments>.

Same thing for the declaration of `iterate_documents()`.